account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -570,16 +635,87 @@ def _default_reconcile_data(self, from_unreconcile=False):
                     self.manual_reference,
                 )
         for line in other_lines:
-            reconcile_auxiliary_id, lines = self._get_reconcile_line(
-                line, "other", from_unreconcile=from_unreconcile
-            )
-            data += lines
+            partial_lines = self._all_partials_lines(line) if from_unreconcile else []
+            if partial_lines:
+                for reconciled_line in (
+                    partial_lines.debit_move_id + partial_lines.credit_move_id - line
+                ):
+                    if (
+                        reconciled_line.move_id.journal_id
+                        == self.company_id.currency_exchange_journal_id
+                    ):
+                        reconcile_auxiliary_id, lines = self._get_reconcile_line(
+                            reconciled_line.move_id.line_ids - reconciled_line,
+                            "other",
+                            from_unreconcile=False,
+                            move=True,
+                        )
+                        data += lines
+                        continue
+                    partial = partial_lines.filtered(
+                        lambda r: r.debit_move_id == reconciled_line
+                        or r.credit_move_id == reconciled_line
+                    )
```

Comment:
```suggestion
                    partial = partial_lines.filtered(
                        lambda r, line=reconciled_line: r.debit_move_id == line
                        or r.credit_move_id == line
                    )
```

Doing something like this will fix the pre-commit